src/main/scala/App/Main.scala

```diff
@@ -2,69 +2,132 @@ package App
 
 import db.UserDB
 import models.UserType.{Admin, Customer}
+import models.{UserType, Users}
 import service.UserRepo
-import models.Users
-import scala.util.{Success,Failure}
 
-import java.util.UUID
+import java.sql.ResultSet
+import scala.annotation.tailrec
 import scala.concurrent.ExecutionContext.Implicits.global
+import scala.io.StdIn
+import scala.util.{Failure, Success, Try}
 
 object Main extends App {
 
   private val userDB = new UserDB
 
   private val userRepo = new UserRepo(userDB)
 
-  private val user1 = Users(UUID.randomUUID(), "jks", 23, "gkp", "12/2/1998", Customer)
-  userRepo.addUser(user1).onComplete{
-    case Success(value) => println(value)
-    case Failure(exception) => println(exception.getMessage)
-  }
+  println("Enter your choice : ")
+  println("1. Add user\n2. Get user detail by Id\n3. Get all users details\n4. Update user detail by Id\n5. Delete user by Id\n6. Delete all\n")
 
-  private val user2 = Users(UUID.randomUUID(), "Jeet", 24, "Delhi", "12/2/1998", Customer)
-  userRepo.addUser(user2).onComplete {
-    case Success(value) => println(value)
-    case Failure(exception) => println(exception.getMessage)
+  private val choice = Try(StdIn.readInt())
+  choice match {
+    case Success(choice) => choice match {
+      case 1 => addUsers()
+      case 2 => getUserDetailById()
+      case 3 => getAllUsers()
+      case 4 => updateUserDetailById()
+      case 5 => deleteUserById()
+      case 6 => deleteAllUsers()
+      case _ => println("Invalid choice!")
+    }
+    case Failure(_) => println("Invalid Choice!")
   }
 
-  private val user3 = Users(UUID.randomUUID(), "Ajit", 24, "Noida", "12/2/1998", Admin)
-  userRepo.addUser(user3).onComplete {
-    case Success(value) => println(value)
-    case Failure(exception) => println(exception.getMessage)
+  private def addUsers(): Unit = {
+    println("Enter name :")
+    val name = StdIn.readLine()
+    println("Enter age :")
+    val age = StdIn.readInt()
+    println("Enter address :")
+    val address = StdIn.readLine()
+    println("Enter a date (YYYY-MM-DD): ")
+    val dob = StdIn.readLine()
+    println("Enter usertype :")
+    val userType = StdIn.readLine()
+
+    val typeOfUser: Either[Unit, UserType] = userType match {
+      case "Admin" => Right(Admin)
+      case "Customer" => Right(Customer)
+      case _ => Left("Invalid user")
```

Comment:
You are returning String here. So the return type should be Either[String, UserType]